ami/socket.go

```diff
@@ -75,13 +79,16 @@ func (s *Socket) Recv(ctx context.Context) (string, error) {
 		select {
 		case msg, ok := <-s.incoming:
 			if !ok {
-				continue
+				return buffer.String(), io.EOF
 			}
 			buffer.WriteString(msg)
 			if strings.HasSuffix(buffer.String(), "\r\n") {
 				return buffer.String(), nil
 			}
+		case err := <-s.error:
+			return buffer.String(), err
```

Comment:
here is where i use the error channel

Reply:
Right, I didn't see that before !
So, the only thing that I would like to change is rename `error` to `errors`

Reply:
done!
ftw, the unit test wont run, hung without any print in console. 
maybe my environment? (i code inside a docker container)